Getting started update (#998)
timescale/docs: timescaledb/getting-started/create-cagg/create-cagg-basics.md

Now that you've been introduced to continuous aggregates, create your own continuous
aggregate from your data. 

## Step 1: Create an aggregate query to use in your continuous aggregate

The data we are using for this tutorial is second-by-second (tick) data for stock trades. 
A popular aggregate pattern used for analyzing stock data is called a [candlestick][candlestick]. 
Generally, candlestick charts use 4 different aggregations over a specific interval 
of time (ie. 1-minute, 5-minute, or 1-day aggregates):

* `high`: highest stock price per interval
* `open`: opening stock price per interval
* `close`: closing stock price per interval
* `low`: lowest stock price per interval 

For this exmaple query, the [`time_bucket()`][time-bucket] function interval will be 1 day. The 
`high` and `low` values can be found by using the PostgreSQL [`MAX()`][max] and [`MIN()`][min] 
functions. Finally, the `open` and `close` values can be found by using the [`first()`][first] 
and [`last()`][last] functions.  

```sql
SELECT
	time_bucket('1 day', "time") AS day,
	symbol,
	max(price) AS high,
	first(price, time) AS open,
	last(price, time) AS close,
	min(price) AS low
FROM stocks_real_time srt

> GROUP BY day, symbol;
> ```

Reviewer: ```suggestion
GROUP BY day, symbol
ORDER BY day DESC, symbol;
```
```
Totally missed this, but we should provide best practices. The other query will be very random on the time column ordering when a user runs it (probably).

Also, I think it would be helpful to see an example couple of rows.

## Step 2: Create continuous aggregate from aggregate query

Now that you have the aggregation query, you can use it to create a continuous aggregate. 

The `CREATE MATERIALIZED VIEW` command triggers the database to create a materialized view with 
the given name, in this case `stock_candlestick_daily`. In the next line, 
`WITH (timescaledb.continuous)` lets the database know that you want to create a continuous 
aggregate and not just a generic materialized view. Then, the `AS` keyword is needed to specify 
the aggregate query you want to use for creating the continuous aggregate. 

To do this, use the following code:

```sql
CREATE MATERIALIZED VIEW stock_candlestick_daily
WITH (timescaledb.continuous) AS
SELECT
	time_bucket('1 day', "time") AS day,
	symbol,
	max(price) AS high,
	first(price, time) AS open,
	last(price, time) AS close,
	min(price) AS low
FROM stocks_real_time srt
GROUP BY day, symbol;
```

> Notice that the `SELECT` statement is the same query you wrote earlier.

Reviewer: ```suggestion
Notice that the `SELECT` statement is the same query you wrote earlier without the `ORDER BY` clause.
```

By default, this code both creates the aggregate *and* materializes the aggregated data.
That means the view is created *and* populated with the aggregate calculations from
your existing hypertable data. 

  <img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/getting-started/continuous-aggregate.jpg" alt="Continuous aggregate upon creation"/>

The query may take some time to run because it needs to perform these calculations. 
After the calculation results are stored, querying the data from the 
continuous aggregate is much faster. Run the following query to get all the data in your 
continuous aggregate, and note how much faster this is than running the aggregate `SELECT` 
query on the raw hypertable data.

```sql

> SELECT * FROM stock_candlestick_daily

Reviewer: ```suggestion
SELECT * FROM stock_candlestick_daily
ORDER BY day DESC, symbol;
```

```

<highlight type="note">
To inspect details about a continuous aggregate, such as its 
configuration or the query used to define it, use the following 
informational view:

```sql

> SELECT * FROM timescaledb_information.continuous_aggregates;

Reviewer: Provide example output IMO

```
</highlight>

> Now that you have your continuous aggregate set up a [continuous aggregate refresh policy][cagg-policy].

Reviewer: ```suggestion
Now that your continuous aggregate is created, the next step is to create a [continuous aggregate refresh policy][cagg-policy].
```

The query may take some time to run because it needs to perform these calculations. 
After the calculation results are stored, querying the data from the 
continuous aggregate is much faster. Run the following query to get all the data in your 
continuous aggregate, and note how much faster this is than running the aggregate `SELECT` 
query on the raw hypertable data.

```sql
SELECT * FROM stock_candlestick_daily
ORDER BY day DESC, symbol;
```

By default, when you query a continuous aggregate, TimescaleDB also includes recent
data that hasn't yet been materialized. In the diagram above, that corresponds to the last
three points of raw data, which belong to an incomplete bucket. This unmaterialized data
is aggregated in real time to provide the most up-to-date results for your query.

<highlight type="note">
To inspect details about a continuous aggregate, such as its 
configuration or the query used to define it, use the following 
informational view:

```sql
SELECT * FROM timescaledb_information.continuous_aggregates;
```

> Results:

Author: ```suggestion
</highlight>

Results:
```

> |hypertable_schema|hypertable_name |view_schema|view_name              |view_owner|materialized_only|compression_enabled|materialization_hypertable_schema|materialization_hypertable_name|view_definition |

Reviewer 2: Table formatting seems to be broken within callouts:
<img width="706" alt="image" src="https://user-images.githubusercontent.com/26616127/163655230-505c8007-65a2-496e-92ae-7349cc93e9b0.png">

|--|--|--|--|--|--|--|--|--|--|
|public           |stocks_real_time|public     |stock_candlestick_daily|tsdbadmin |false            |false              |_timescaledb_internal            |_materialized_hypertable_11    | SELECT time_bucket('1 day'::interval, srt."time") AS day,¶    srt.symbol,¶    max(srt.price) AS high,¶    first(srt.price, srt."time") AS open,¶    last(srt.price, srt."time") AS close,¶    min(srt.price) AS low¶   FROM stocks_real_time srt¶  GROUP BY (t|

> </highlight>

Author: ```suggestion
```

The query may take some time to run because it needs to perform these calculations. 
After the calculation results are stored, querying the data from the 
continuous aggregate is much faster. Run the following query to get all the data in your 
continuous aggregate, and note how much faster this is than running the aggregate `SELECT` 
query on the raw hypertable data.

```sql
SELECT * FROM stock_candlestick_daily
ORDER BY day DESC, symbol;
```

By default, when you query a continuous aggregate, TimescaleDB also includes recent
data that hasn't yet been materialized. In the diagram above, that corresponds to the last
three points of raw data, which belong to an incomplete bucket. This unmaterialized data
is aggregated in real time to provide the most up-to-date results for your query.

<highlight type="note">
To inspect details about a continuous aggregate, such as its 
configuration or the query used to define it, use the following 
informational view:

```sql
SELECT * FROM timescaledb_information.continuous_aggregates;
```
</highlight>

Results:

> |hypertable_schema|hypertable_name |view_schema|view_name              |view_owner|materialized_only|compression_enabled|materialization_hypertable_schema|materialization_hypertable_name|view_definition|
> |-|-|-|-|-|-|-|-|-|-|
> |public|stocks_real_time|public|stock_candlestick_daily|tsdbadmin |false|false|_timescaledb_internal        |_materialized_hypertable_11|SELECT time_bucket('1 day'::interval, srt."time") AS day,srt.symbol, max(srt.price) AS high, first(srt.price, srt."time") AS open, last(srt.price, srt."time") AS close, min(srt.price) AS low FROM stocks_real_time srt GROUP BY day, srt.symbol|

Reviewer: ```suggestion
```bash
|hypertable_schema|hypertable_name |view_schema|view_name              |view_owner|materialized_only|compression_enabled|materialization_hypertable_schema|materialization_hypertable_name|view_definition|
|-|-|-|-|-|-|-|-|-|-|
|public|stocks_real_time|public|stock_candlestick_daily|tsdbadmin |false|false|_timescaledb_internal        |_materialized_hypertable_11|SELECT time_bucket('1 day'::interval, srt."time") AS day,srt.symbol, max(srt.price) AS high, first(srt.price, srt."time") AS open, last(srt.price, srt."time") AS close, min(srt.price) AS low FROM stocks_real_time srt GROUP BY day, srt.symbol|
```
```

# Creating continuous aggregates

Now that you've been introduced to continuous aggregates, create your own continuous
aggregate from your data. 

## Create an aggregate query to use in your continuous aggregate

The data we are using for this tutorial is second-by-second (tick) data for stock trades. 
A popular aggregate pattern used for analyzing stock data is called a [candlestick][candlestick]. 
Generally, candlestick charts use 4 different aggregations over a specific interval 
of time (for example, 1-minute, 5-minute, or 1-day aggregates):

* `high`: highest stock price per interval
* `open`: opening stock price per interval
* `close`: closing stock price per interval
* `low`: lowest stock price per interval 

For this example query, the [`time_bucket()`][time-bucket] interval is 1 day. The 
`high` and `low` values can be found by using the PostgreSQL [`MAX()`][max] and [`MIN()`][min] 
functions. Finally, the `open` and `close` values can be found by using the [`first()`][first] 
and [`last()`][last] functions.  

<procedure>

> 

Reviewer 2: ```suggestion

### Creating an aggregate query

```

	ORDER BY day DESC, symbol;
	```
	**Results:**
	```bash
	day                          |symbol|high    |open    |close   |low     |
	-----------------------------+------+--------+--------+--------+--------+
	2022-05-03 20:00:00.000 -0400|AAPL  |164.9799|  159.32| 164.545|  159.25|
	2022-05-03 20:00:00.000 -0400|ABBV  |   151.7|  150.99|  151.32|  147.59|
	2022-05-03 20:00:00.000 -0400|ABNB  |158.7158|  148.84|  153.58|  145.88|
	2022-05-03 20:00:00.000 -0400|ABT   |   115.2|  111.64|  115.08|  111.14|
	2022-05-03 20:00:00.000 -0400|ADBE  |  421.93|  407.61|  419.53|  395.06|
	2022-05-03 20:00:00.000 -0400|AMAT  |  118.47| 114.279|  117.95|  112.04|
	```

</procedure>

## Create continuous aggregate from aggregate query

Now that you have the aggregation query, you can use it to create a continuous aggregate. 

The `CREATE MATERIALIZED VIEW` command triggers the database to create a materialized view with 
the given name, in this case `stock_candlestick_daily`. In the next line, 
`WITH (timescaledb.continuous)` instructs TimescaleDB to create a continuous 
aggregate and not just a generic materialized view. Finally, the query from above
is added after the `AS` keyword. 

<procedure>

> 

Reviewer 2: ```suggestion

### Creating a continuous aggregate from an aggregate query

```

The query may take some time to run because it needs to perform these calculations
across all of your stock trade data the first time. After the calculation results 
are stored, querying the data from the continuous aggregate is much faster. 

Run the following query to get all the data in your continuous aggregate, and note 
how much faster this is than running the aggregate `SELECT` query on the raw hypertable data.

```sql
SELECT * FROM stock_candlestick_daily
ORDER BY day DESC, symbol;
```

## Real-time continuous aggregates
By default, all continuous aggregates are created as *real-time* aggregates. This
means that TimescaleDB always appends (or `UNION`) recent data that has not yet been materialized
through a refresh policy to the output of the continuous aggregate. In the diagram 
above that corresponds to the last three points of raw data, which belong to an 
incomplete bucket. 

<highlight type="note">
To inspect details about a continuous aggregate, such as its 
configuration or the query used to define it, use the following 
informational view:

```sql
SELECT * FROM timescaledb_information.continuous_aggregates;
```
**Results:**
```bash

> hypertable_schema|hypertable_name |view_schema|view_name     |view_owner|materialized_only|compression_enabled|materialization_hypertable_schema|materialization_hypertable_name|view_definition                                                                                                                                                                                                                                                |

Reviewer 2: Might look better displayed with `\x` on. Written in one long line, the query is quite hard to read.

```sql
SELECT * FROM timescaledb_information.continuous_aggregates;
```
</highlight>

Results:

> |hypertable_schema|hypertable_name |view_schema|view_name              |view_owner|materialized_only|compression_enabled|materialization_hypertable_schema|materialization_hypertable_name|view_definition                                    |
> |--|--|--|--|--|--|--|--|--|-----------------------------------------------------|
> |public           |stocks_real_time|public     |stock_candlestick_daily|tsdbadmin |false            |false              |_timescaledb_internal            |_materialized_hypertable_11    | SELECT time_bucket('1 day'::interval, srt."time") AS day,srt.symbol, max(srt.price) AS high, first(srt.price, srt."time") AS open, last(srt.price, srt."time") AS close, min(srt.price) AS low FROM stocks_real_time srt GROUP BY day, srt.symbol|

Reviewer 3: ```suggestion
|hypertable_schema|hypertable_name |view_schema|view_name              |view_owner|materialized_only|compression_enabled|materialization_hypertable_schema|materialization_hypertable_name|view_definition|
|-|-|-|-|-|-|-|-|-|-|
|public|stocks_real_time|public|stock_candlestick_daily|tsdbadmin |false|false|_timescaledb_internal        |_materialized_hypertable_11|SELECT time_bucket('1 day'::interval, srt."time") AS day,srt.symbol, max(srt.price) AS high, first(srt.price, srt."time") AS open, last(srt.price, srt."time") AS close, min(srt.price) AS low FROM stocks_real_time srt GROUP BY day, srt.symbol|
```

I think? It's really hard to do this in the GH interface 😂

Results:

> ```bash

Reviewer 3: Is this supposed to be here? The table won't render if it's inside a code block.
